### PR TITLE
Reporting dashboard v2

### PR DESCRIPTION
**Pull request by OptimusRhine**
https://www.pivotaltracker.com/story/show/150063689

* added Usage Statistics section to Reporting Dashboard
  * includes registrations and logins (incl. intervention breakdowns)
  * currently displays by day, month, year, and all time
    * eventual implementation will be a filterable graph - this is more a data POC
* added Encounter and user registration data collection to backend, for new Usage Stats section
* added 'Unspecified' option for orgs, for users with no orgs yet (different from users with 'none of the above' org)

### CODE REVIEW

File: portal/views/portal.py [hunk 1]
```diff
@@ -1036,8 +1039,20 @@ def get_reporting_counts():
             counts['interventions'][interv.description] += 1
             if (any(doc.intervention == interv for doc in user.documents)):
                 counts['intervention_reports'][interv.description] += 1
-        for org in user.organizations:
-            counts['organizations'][org.name] += 1
+        if not user.organizations:
+            counts['organizations']['Unspecified'] += 1
```
Comment (pbugni) [inline portal/views/portal.py]: Hmm, we have "none of the above" in play elsewhere.  See organizations.id == 0.

File: portal/views/portal.py [hunk 2]
```diff
@@ -1036,8 +1039,20 @@ def get_reporting_counts():
...
+        else:
+            for org in user.organizations:
+                counts['organizations'][org.name] += 1
+        counts['registrations'].append(user.registered)
+
+    for enc in Encounter.query.filter_by(auth_method='password_authenticated',
+                                         status='finished'):
```
Comment (pbugni) [inline portal/views/portal.py]: FWIW, we have a bug around `finished` encounters, which will get cleaned up when Ivan's [Session Monitor](https://github.com/uwcirg/true_nth_usa_portal/pull/804) solution is ready, that is, encounters only get marked finished if the browser session isn't killed.

You might ignore the `status` for now.

File: portal/views/portal.py (hunk 1 above)
Comment (OptimusRhine) [inline portal/views/portal.py]: Yep, that 'none of the above' org is already taken into account when looking at the `User.organizations`. This is to handle cases where the user has not yet been assigned to _any_ orgs, _including_ the 'none of the above' org; see Justin's comments in [the PT story here](https://www.pivotaltracker.com/story/show/150063689).

Comment (pbugni) [inline portal/views/portal.py]: Gotcha.  Comments always help when such a situation isn't obvious - but now that I've read that, it is.

**Pull request merged**